src/ansys/mapdl/core/math.py

```diff
@@ -1399,10 +1399,19 @@ def __repr__(self):
         return f"APDLMath Vector Size {self.size}"
 
     def __getitem__(self, num):
+        info = self._mapdl._data_info(self.id)
+        dtype = ANSYS_VALUE_TYPE[info.stype]
         if num < 0:
             raise ValueError("Negative indices not permitted")
+
         self._mapdl.run(f"pyval={self.id}({num+1})", mute=True)
-        return self._mapdl.scalar_param("pyval")
+        item_val = self._mapdl.scalar_param("pyval")
+
+        if MYCTYPE[dtype] == "C" or MYCTYPE[dtype] == "Z":
+            self._mapdl.run(f"pyval_img={self.id}({num+1},2)", mute=True)
+            img_val = self._mapdl.scalar_param("pyval_img")
+            item_val = item_val + img_val * 1j
+        return item_val
```

Comment:
```suggestion
        item_val = self._mapdl.scalar_param("pyval_")

        if MYCTYPE[dtype] == "C" or MYCTYPE[dtype] == "Z":
            self._mapdl.run(f"pyval_img_={self.id}({num+1},2)", mute=True)
            img_val = self._mapdl.scalar_param("pyval_img_")
            
            # Clean parameters
            self._mapdl.run("item_val =")
            self._mapdl.run("pyval_img_=")

        return item_val + img_val * 1j
```

We should try to clean up after running the code. Also use trailing ``_`` for noticing it is an internal parameter.

Reply:
If we are returning `return item_val + img_val * 1j` can I initialize `img_val ` to 0 initially.
Could hit an error when it does not go into the condition of dtype = "C" / "Z"